mlflow/xgboost/__init__.py

```diff
@@ -361,8 +362,34 @@ def predict(
         Returns:
             Model predictions.
         """
+        import xgboost as xgb
+
         predict_fn = _wrapped_xgboost_model_predict_fn(self.xgb_model)
-        return predict_fn(dataframe, **(params or {}))
+        params = params or {}
+        # filter is applied inside predict_fn wrapper for xgb.Booster
+        if not isinstance(self.xgb_model, xgb.Booster):
+            params = _exclude_unrecognized_kwargs(predict_fn, params)
+        return predict_fn(dataframe, **params)
+
+
+def _exclude_unrecognized_kwargs(predict_fn, kwargs):
+    filtered_kwargs = {}
+    allowed_params = inspect.signature(predict_fn).parameters
+    # avoid excluding kwargs when predict function uses args or kwargs
+    if not allowed_params.keys().isdisjoint({"args", "kwargs"}):
+        return kwargs
+    invalid_params = set()
+    for key, value in kwargs.items():
+        if key in allowed_params:
+            filtered_kwargs[key] = value
+        else:
+            invalid_params.add(key)
+    if invalid_params:
+        _logger.warning(
+            f"Params {invalid_params} are not accepted by the xgboost model, "
+            "ignoring them during predict."
+        )
+    return filtered_kwargs
```

Comment:
/ai we want to skip parameter exclusion if the function has `*args` or `**kwargs`, `args` or `kwargs`. Please update the code.